projectroles/forms.py

```diff
@@ -541,9 +554,83 @@ def __init__(self, project=None, current_user=None, *args, **kwargs):
             self.fields['description'].widget = forms.HiddenInput()
             self.fields['readme'].widget = forms.HiddenInput()
 
+    @classmethod
+    def _validate_app_settings(
+        self,
+        cleaned_data,
+        app_plugins,
+        app_settings,
+        p_kwargs,
+        instance,
+        instance_owner_as,
+    ):
+        """Validate and clean app_settings form fields"""
+        errors = []
+
+        if cleaned_data.get('type') == PROJECT_TYPE_CATEGORY:
+            return (cleaned_data, errors)
+
+        for plugin in app_plugins + [None]:
+            if plugin:
+                name = plugin.name
+                p_settings = app_settings.get_definitions(
+                    APP_SETTING_SCOPE_PROJECT, plugin=plugin, **p_kwargs
+                )
+                app_settings_errors = plugin.validate_form_app_settings(
```

Comment:
Here, you provide the definitions of app settins to the validation method. However, what we actually need to provide is the list of actual settings and their values, from `cleaned_data`. Also, only the settings of the specific app plugin should be provided to the appropriate plugin.

Of course, the plugin already knows about its own definitiions, so those do not have to be provided to this method.